### PR TITLE
bug: controller-manager's os.Exit will not wait for the defer function

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of # https://github.com/karmada-io/karmada/issues/1313

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: cmd/controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -85,21 +87,18 @@ func Run(ctx context.Context, opts *options.Options) error {
 		LivenessEndpointName:       "/healthz",
 	})
 	if err != nil {
-		klog.Errorf("failed to build controller manager: %v", err)
-		return err
+		return fmt.Errorf("failed to build controller manager: %w", err)
```
Comment (RainbowMango) [inline cmd/controller-manager/app/controllermanager.go:90]: I can't see any benefits from the wrapped error here, but if the error happens it becomes a little complicated to figure out where the problem is.

I suggest letting this PR focus on what it meant to do.

Comment (RainbowMango) [reply cmd/controller-manager/app/controllermanager.go:90]: cc @lonelyCZ

Comment (kerthcet) [reply cmd/controller-manager/app/controllermanager.go]: thanks, that also makes sense to me. Updated.

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

**Comment by kerthcet** [comment]
Would like to apply this change to others.

**Pull request merged**